src/parser.rs

```diff
@@ -6318,6 +6317,31 @@ impl<'a> Parser<'a> {
     pub fn index(&self) -> usize {
         self.index
     }
+
+    fn check_on_update_expr_is_valid(expr: &Expr) -> Result<(), ParserError> {
```

Comment:
my preference is to leave this type of "semantic" check downstream -- so I would prefer that sqlparser accepts any expression in this place, and then downstream crates can enforce limits like the expr must be one of the specified time functions. 

@AugustoFKL  and I tried to explain this difference in the readme recently

https://github.com/sqlparser-rs/sqlparser-rs#extensible-sql-lexer-and-parser-for-rust

> This crate provides only a syntax parser, and tries to avoid applying
> any SQL semantics, and accepts queries that specific databases would
> reject, even when using that Database's specific `Dialect`. For
> example, `CREATE TABLE(x int, x int)` is accepted by this crate, even
> though most SQL engines will reject this statement due to the repeated
> column name `x`.

> This crate avoids semantic analysis because it varies drastically
> between dialects and implementations. If you want to do semantic
> analysis, feel free to use this project as a base